utils/regr/moda.h

```diff
@@ -105,30 +151,31 @@ struct ModaData : public UserData
   virtual void unserialize(messageqcpp::ByteStream& bs);
 
   template <class T>
-  std::unordered_map<T, uint32_t, hasher<T> >* getMap()
+  std::unordered_map<T, uint32_t, hasher<T>, comparator<T> >* getMap()
   {
     if (!fMap)
     {
       // Just in time creation
-      fMap = new std::unordered_map<T, uint32_t, hasher<T> >;
+      fMap = new std::unordered_map<T, uint32_t, hasher<T>, comparator<T> >(
```

Comment:
Could you ellaborate on what is this map for? If it will contain up to 100 entries flatmap might be more interesting in terms of memory consumption.

Reply:
This map keys on the unique values of the result set with their respective counts. The result set can be 0 to infinity entries, limited by memory.

The algorithm is: for each value sent to the function from the query, look up that value in the map and increase it's count. If it isn't found, add with a count of 1. At the end, look for the entry with the largest count. If needed, apply tie-breakers.